components/dashboard/src/workspaces/start-workspace-modal-context.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import React, { createContext, useEffect, useState } from 'react';
+
+export const StartWorkspaceModalContext = createContext<{
+    isStartWorkspaceModalVisible?: boolean,
+    setIsStartWorkspaceModalVisible: React.Dispatch<boolean>,
+}>({
+    setIsStartWorkspaceModalVisible: () => null,
+});
+
+export const StartWorkspaceModalContextProvider: React.FC = ({ children }) => {
+    const [ isStartWorkspaceModalVisible, setIsStartWorkspaceModalVisible ] = useState<boolean>(false);
+
+    useEffect(() => {
+        const onKeyDown = (event: KeyboardEvent) => {
+            if ((event.metaKey || event.ctrlKey) && event.key === 'o') {
+                event.preventDefault();
+                setIsStartWorkspaceModalVisible(true);
+            }
+        };
+        window.addEventListener('keydown', onKeyDown);
+        return () => {
+            window.removeEventListener('keydown', onKeyDown);
+        }
+    }, []);
+
+    return <StartWorkspaceModalContext.Provider value={{ isStartWorkspaceModalVisible, setIsStartWorkspaceModalVisible }}>
+        {children}
+    </StartWorkspaceModalContext.Provider>;
+}
+
+export const StartWorkspaceModalKeyBinding = `${/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform) ? '⌘' : 'Ctrl+'}O`;
```

Comment:
@gtsiolis `Ctrl+O` is the keybinding for non-Mac users (i.e. Windows / Linux). From your reference:

> Ctrl+X | Current Microsoft notation
> ⌃X | Classic Mac OS and macOS notation

I guess `^O` will speak mostly to Mac users. I remember seeing this notation before switching to a Mac and being very confused myself (especially since my keyboard didn't have the `^` symbol on the right key).

Also, it's worth noting that VS Code chose to use `⌘O` on Mac and `Ctrl+O` on Windows (not `^O`). I guess aligning with VS Code is good here. 💭

Reply:
Wait, I could but wrong isn't `^` a notation for the `Ctrl` key? 💭 

Re-posting references from https://github.com/gitpod-io/gitpod/pull/7715#discussion_r793649559:
[1]: https://en.wikipedia.org/wiki/Control_key#Notation
[2]: https://en.wikipedia.org/wiki/Caret_notation

Reply:
> Wait, I could be wrong but isn't `^` a notation for the `Ctrl` key? 💭 

Indeed, it seems to be, but my point is, only Mac OS users typically know it, while Windows and Linux users will very likely not recognize this notation.